linode/manager
# feat: [M3-8335] - Link Account Limit ticket in more flows with error notice

## Description 📝
#10620 focused on the creation of the new account limit support ticket, linking to it from the Linode Create flow, and providing test coverage.

This ticket will link to the the account limit support ticket from the "contact Support" text on the Volumes, NodeBalancers, Firewall, Kubernetes (LKE create and Add a Node Pool), and Database create flows. Currently, the link in these flows takes users to a general support ticket.

## Changes  🔄
- Removes the global intercept on the text "contact Support" and "open a support ticket"
- Links to an account limit ticket for Create flows listed above when the expected error text is present in the error message
- Fixes a bug in a `handleGeneralError` Formik util that didn't account for the APIError reason being a string *or* JSX

## Target release date 🗓️
8/5/24

## Preview 📷

<details>
<summary>
Videos 🎥 
</summary>

| Flow | Limit Error |
| ------- | ------- | 
| Volumes | <video src="https://github.com/user-attachments/assets/5fdb4771-b4b6-4928-82db-bc7b47d3595d" />  | 
| NodeBalancers | <video src="https://github.com/user-attachments/assets/2fb32a68-1ac5-47ad-b302-db53e0f8a760" />  | 
| LKE Create - Thing Limit | <video src="https://github.com/user-attachments/assets/61288ed6-6eef-4959-874e-89042492d456" />  |
| LKE Create - Rep Limit | <video src="https://github.com/user-attachments/assets/10dc6db1-be18-4d47-b3bb-46bd1ef51dd0" /> | 
| LKE Add Node Pool - Thing Limit | <video src="https://github.com/user-attachments/assets/c4aebdf4-fc4d-48cf-bfae-de4ceb59b3a7" />  |
| LKE Add Node Pool - Rep Limit | <video src="https://github.com/user-attachments/assets/c4aebdf4-fc4d-48cf-bfae-de4ceb59b3a7" />  | <video src="https://github.com/user-attachments/assets/46e04253-41b7-4267-9d4d-117e38a15be4" /> | 
| Firewalls | <video src="https://github.com/user-attachments/assets/1befcbfe-0086-4d80-a27d-d7580cd5e286" />  | 
| Databases | <video src="https://github.com/user-attachments/assets/6f8d5543-3603-407f-bb7a-d203804cc866" /> | 

</details>

## How to test 🧪

### Prerequisites
(How to setup test environment)
- **Please test this on your dev account so we don't bother Customer Support with real tickets.**
- Check out this PR and switch your environment to dev.
- Log into admin and be ready to change some things on your dev account.

### Verification steps
(How to verify changes)
- In admin, set your thing limit to 0.
- For each of the above create flows (Volumes, NBs, LKE Create + Add a Node Pool, Firewalls, Databases):
- Fill out the create form and try to create the entity. 
- Confirm that the UI displays an error notice when user has reached a limit on their account: `You’ve reached a limit for the number of active services on your account. Please contact Support to request an increase and provide the total number of services you may need.`
- Click on the "contact Support" link and confirm that you are redirected to the Account Limit ticket and the expected fields show.
   - The Linode plan field should only show for Linode and LKE account limit tickets.
   - The correct entity should be shown in the "total entities" field, with the correct number of existing entities from your account as helper text.
- For LKE only, setting your reputation to 0 should result in the error: `A limit on your account is preventing the deployment of the selected service. To request access, please contact Support and provide the service name.` You should also link to an account limit support ticket.

- Verify that this test isn't flaking in CI:
```
yarn cy:run -s "cypress/e2e/core/helpAndSupport/open-support-ticket.spec.ts"
```

## As an Author I have considered 🤔

*Check all that apply*
